synapse/tests/test_lib_stormtypes.py

```diff
@@ -188,6 +188,58 @@ def check_fire_mesgs(storm_mesgs, expected_data):
             self.eq(erfo[1][0], 'NoSuchName')
             self.eq(erfo[1][1].get('name'), 'newp')
 
+    async def test_storm_lib_query(self):
+        async with self.getTestCore() as core:
+            # basic
+            q = '''
+            $foo = ${ [test:str=theevalthatmendo] }
+            $foo.exec()
+            '''
+            await core.nodes(q)
+            nodes = await core.nodes('test:str=theevalthatmendo')
+            self.len(1, nodes)
+
+            # make sure our scope goes down
+            q = '''
+            $bar = ${ [test:str=$foo] }
+
+            $foo = "this little node went to market"
+            $bar.exec()
+            $foo = "this little node stayed home"
+            $bar.exec()
+            $foo = "this little node had roast beef"
+            $bar.exec()
+            '''
+            msgs = await core.streamstorm(q).list()
+            nodes = [m for m in msgs if m[0] == 'node:add']
+            self.len(3, nodes)
+            self.eq(nodes[0][1]['ndef'], ('test:str', 'this little node went to market'))
+            self.eq(nodes[1][1]['ndef'], ('test:str', 'this little node stayed home'))
+            self.eq(nodes[2][1]['ndef'], ('test:str', 'this little node had roast beef'))
+
+            # but that it doesn't come back up
+            q = '''
+            $foo = "that is one neato burrito"
+            $baz = ${ $bar=$lib.str.concat(wompwomp, $lib.guid()) }
+            $baz.exec()
+            $lib.print($bar)
+            [ test:str=$foo ]
+            '''
+
+            msgs = await core.streamstorm(q).list()
+            prints = [m for m in msgs if m[0] == 'print']
+            self.len(0, prints)
+
+            # make sure returns
+            q = '''
+            $foo = $(10)
+            $bar = ${ return ( $($foo+1) ) }
+            [test:int=$bar.exec()]
+            '''
+            msgs = await core.streamstorm(q).list()
+            nodes = [m for m in msgs if m[0] == 'node']
+            self.len(1, nodes)
```

Comment:
What is the expected value of this nodes ndef?

Reply:
Added :+1: